### DISCUSSION TITLE
dispatchBrowserEvent supported ?

### DISCUSSION THREAD

**Discussion by anglin-rogersoft** [rappasoft/laravel-livewire-tables#1358] [Q&A]
I need to dispatch a browser event. Is it supported ?
@lrljoe

**Comment by lrljoe** [comment]
Thanks for the tag.

It depends what you're trying to dispatch an event from!  You can certainly attach various attributes across the table, and can use custom views for columns, in which you can run any JS that you like etc.

What's the use case exactly?

**Reply by anglin-rogersoft** [reply]
Use Case :- 
Usually I use this for showing notifications ( sweet alert 2 ).

I place a window event listener in footer page & in case of notification I dispatch a browser event. It's very handy in my opinion 😉.

In Footer : 
` window.addEventListener('swal:notification',function(e){ 
          Toast.fire({
          icon: e.detail.icon,
          title: e.detail.title
          });
      }); `

On livewire component :
` $this->dispatchBrowserEvent('swal:notification', [
            'icon' => 'success',
            'title' => 'Course deleted successfully',
        ]); `


// Not only notifications, some live interactions such as datatable data update & tabs items counts .. ( unlimited possibilities ) without  network renderings I can render so much things  😍

**Reply by lrljoe** [reply]
Yep, you can definitely do this, depending on whether you're using v2 or v3, there's some slightly different approaches you can take, let me know which version you're on, and I'll advise.

With v3 (LW3), the better approach is doing it all client-side, but you can also pass parameters to a function in your Table Component and dispatch the event.

With v2 (LW2), it is smoother to always call a public function in your Table Component, with relevant parameters, and emit an event.

Feel free to prod me on the official Discord, as I'm tagged in 99% of the threads on here, so someone poking me on Discord typically gets a faster response.
